chessington/engine/pieces.py

```diff
@@ -49,7 +83,36 @@ class Knight(Piece):
     """
 
     def get_available_moves(self, board):
-        return []
+        available_moves = []
+        current_square = board.find_piece(self)
+        possible_squares = self.get_possible_squares(board, current_square)
+
+        for square in possible_squares:
+            if board.get_piece(square) is None:
+                available_moves.append(square)
+            elif board.get_piece(square).player != self.player:
+                available_moves.append(square)
+
+        return available_moves
+
+    def get_possible_squares(self, board, current_square):
+        possible_squares = []
+        possible_squares.append(Square(current_square.row - 1, current_square.col - 2))
+        possible_squares.append(Square.at(current_square.row + 1, current_square.col - 2))
+        possible_squares.append(Square.at(current_square.row + 2, current_square.col - 1))
+        possible_squares.append(Square.at(current_square.row + 2, current_square.col + 1))
+        possible_squares.append(Square.at(current_square.row + 1, current_square.col + 2))
+        possible_squares.append(Square.at(current_square.row - 1, current_square.col + 2))
+        possible_squares.append(Square.at(current_square.row - 2, current_square.col + 1))
+        possible_squares.append(Square.at(current_square.row - 2, current_square.col - 1))
+        return self.get_legal_squares(board, possible_squares)
```

Comment:
You can use the same idea you had for `Bishop` and `Rook` with the `tuple` here too.